### PR TITLE
Fix for missing presence notifications

### PR DESCRIPTION
**Pull request by or-else**
Here is the fix.

### CODE REVIEW

File: server/topic.go [hunk 1]
```diff
@@ -2580,24 +2586,47 @@ func (t *Topic) mostRecentSession() *Session {
 	return sess
 }
 
-func (t *Topic) suspend() {
-	atomic.StoreInt32((*int32)(&t.suspended), 1)
+const (
+	// Topic is just created.
+	topicStatusNew = 0
+	// Topic is fully initialized.
+	topicStatusLoaded = 1
```
Comment (aforge) [inline server/topic.go]: I understand `topicStatusLoaded` is only meaningful in ME, P2P and GRP contexts.
FND and SYS topics will forever remain in `topicStatusNew` state.
Is that the intent?
If so, perhaps, SYS and FND should go to `topicStatusLoaded` right away?

File: server/topic.go [hunk 2]
```diff
@@ -782,39 +784,12 @@ func (t *Topic) handleSubscription(h *Hub, sreg *sessionJoin) error {
 	return nil
 }
 
-// Send presence notification in response to a subscription
-func (t *Topic) sendSubNotifications(asUid types.Uid, sreg *sessionJoin) {
+// Send immediate presence notification in response to a subscription.
+// These notifications are always sent immediately even if background is requested.
```
Comment (aforge) [inline server/topic.go]: This defeats the purpose of introducing background requests.
We don't care about presence notifications when background a request is sent. Should we discuss this further?

Comment (or-else) [reply server/topic.go]: I split the notifications into two sets: sent immediately regardless of the background status and those optionally delayed. The always immediate notifications are p2p & group notifications on a new subscription. I think it's a weird case to join a new chat in background.

File: server/topic.go (hunk 1 above)
Comment (or-else) [reply server/topic.go]: done

File: server/topic.go (hunk 2 above)
Comment (aforge) [reply server/topic.go]: Okay, thanks for explaining.
This is good.

**Pull request merged**